src/ssl.c

```diff
@@ -48370,6 +48370,357 @@ int wolfSSL_PEM_write_bio_PKCS7(WOLFSSL_BIO* bio, PKCS7* p7)
+            if (ret) return ret;
+            if (mimeType == MIME_HEADER) {
+                nextHeader->name = nameAttr;
+                nextHeader->body = bodyVal;
+                nextHeader->next = curHeader;
+                curHeader = nextHeader;
+                nextHeader = XMALLOC(sizeof(MimeHeader), NULL, DYNAMIC_TYPE_PKCS7);
+                if (!nextHeader) return MEMORY_E;
+                XMEMSET(nextHeader, 0, sizeof(MimeHeader));
+            } else {
+                nextParam->attribute = nameAttr;
+                nextParam->value = bodyVal;
+                nextParam->next = curHeader->parameters;
+                curHeader->parameters = nextParam;
+                nextParam = XMALLOC(sizeof(MimeParameter), NULL, DYNAMIC_TYPE_PKCS7);
+                if (!nextParam) return MEMORY_E;
+                XMEMSET(nextParam, 0, sizeof(MimeParameter));
+            }
+            mimeStatus = MIME_NAMEATTRIB;
+        }
+
+        curLine += lineLen;
+        lineLen = wolfSSL_BIO_gets(in, curLine, inLen);
+        if (lineLen<=0) return ASN_PARSE_E;
+        inLen -= lineLen;
+        /* Line with just newline signals end of headers. */
+        if ((lineLen==2 && !XSTRNCMP(curLine, "\r\n", 2)) ||
+            (lineLen==1 && (curLine[0] == '\r' || curLine[0] == '\n')))
+            isEnd = 1;
+        mimeStatus = MIME_NAMEATTRIB;
+    }
+
+    *headers = curHeader;
+    if (lines) XFREE(lines, NULL, DYNAMIC_TYPE_PKCS7);
+    if (nextHeader) XFREE(nextHeader, NULL, DYNAMIC_TYPE_PKCS7);
+    if (nextParam) XFREE(nextParam, NULL, DYNAMIC_TYPE_PKCS7);
+
+    return 0;
+}
+
+int MIME_header_strip(char* in, char** out, word32 start, word32 end)
+{
+    word32 inPos = start;
+    word32 outPos = 0;
+    word32 inLen = (word32)XSTRLEN(in);
+
+    if (end < start || !in || start > inLen || end > inLen) return ASN_PARSE_E;
+
+    *out = XMALLOC(((end-start)+2)*sizeof(char), NULL, DYNAMIC_TYPE_PKCS7);
+    if (*out == NULL) return MEMORY_E;
+
+    while (inPos <= end) {
+        if (in[inPos] >= MIME_HEADER_ASCII_MIN && in[inPos] <= MIME_HEADER_ASCII_MAX
+            && in[inPos] != ';' && in[inPos] != '\"') {
+            (*out)[outPos] = in[inPos];
+            outPos++;
+        }
+        inPos++;
+    }
+    (*out)[outPos] = '\0';
+
+    return 0;
+}
+
+MimeHeader* MIME_find_header_name(const char* name, MimeHeader* header)
+{
+    size_t len = XSTRLEN(name);
+
+    while (header) {
+        if (!XSTRNCMP(name, header->name, len)) {
+            return header;
+        }
+        header = header->next;
+    }
+
+    return header;
+}
+
+MimeParameter* MIME_find_param_attr(const char* attribute, MimeParameter* param)
+{
+    size_t len = XSTRLEN(attribute);
+
+    while (param) {
+        if (!XSTRNCMP(attribute, param->attribute, len)) {
+            return param;
+        }
+        param = param->next;
+    }
+
+    return param;
+}
+
+void MIME_free_headers(MimeHeader* head)
```

Comment:
Please add comment explaining function, args, return values.  Also, please add return value in case needed later.  Can simply return 0 for now for success.